templates/r10k-configmap.yaml

```diff
@@ -2,66 +2,59 @@
 apiVersion: v1
 kind: ConfigMap
 metadata:
-  name: r10k-config
+  name: r10k-code-config
   labels:
     {{- include "puppetserver.r10k.labels" . | nindent 4 }}
 data:
   r10k.yaml: |
     # The location to use for storing cached Git repos
-    :cachedir: '/var/tmp/r10k_cache'
-
+    :cachedir: '/etc/puppetlabs/code/r10k_cache'
     # A list of git repositories to create
     :sources:
       # This will clone the git repository and instantiate an environment per
       # branch in '/etc/puppetlabs/code/environments'
       :puppet_repo:
         remote: '{{.Values.puppetserver.puppeturl}}'
         basedir: '/etc/puppetlabs/code/environments'
-      {{- if .Values.hiera.hieradataurl }}
+    :git:
+      provider: 'rugged' # Either 'shellgit' or 'rugged', defaults to 'shellgit'
+    {{- with .Values.r10k.code.viaSsh.credentials }}
+    {{- if or (.existingSecret) (and (.ssh.value) (.known_hosts.value)) }}
+      private_key: '/root/.ssh/id_rsa'
+    {{- end }}
+    {{- end }}
+      repositories:
+        - remote: '{{.Values.puppetserver.puppeturl}}'
+{{- end }}
+---
```

Comment:
Should be in separate files. The best practice is to keep different K8s objects defined in their own YAML file - in case of an issue the debugging is easier.